malloc/malloc.c

```diff
@@ -13,7 +13,9 @@
 */
 char *string_dup(char *src)
 {
-
+    char *duplicate = (char *) malloc(strlen(src));
```

Comment:
Is there a need to cast the malloc'd memory to a `char *` here? I haven't seen that technique before.